src/main/java/example/basic/CombinedFieldQueryExample.java

```diff
@@ -0,0 +1,130 @@
+//
+// In the following example, we demonstrate how to use `CombinedFieldQuery` to perform a search over the fields `year`, `make`, and `model`.
+
+package example.basic;
+
+import org.apache.lucene.analysis.standard.StandardAnalyzer;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.TextField;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexWriter;
+import org.apache.lucene.index.IndexWriterConfig;
+import org.apache.lucene.sandbox.search.CombinedFieldQuery;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.store.ByteBuffersDirectory;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.util.BytesRef;
+
+import java.io.IOException;
+
+public class CombinedFieldQueryExample {
+    public static void main(String[] args) throws Exception {
+        // # Step 1: Create a new index
+        // In-memory ByteBuffersDirectory is used for indexing
+        Directory directory = new ByteBuffersDirectory();
+
+        // Create the configuration for the index writer, using a standard analyzer
+        IndexWriterConfig config = new IndexWriterConfig(new StandardAnalyzer());
+
+        // Create an IndexWriter to add documents to the index
+        IndexWriter indexWriter = new IndexWriter(directory, config);
+
+        // # Step 2: Add documents to the index
+        // Add sample documents with year (as a string), make, and model fields
+        addDocument(indexWriter, "2011", "Toyota", "Corolla");
+        addDocument(indexWriter, "2015", "Honda", "Civic");
+        addDocument(indexWriter, "2020", "Toyota", "Camry");
+
+        // Close the IndexWriter after adding all documents
+        indexWriter.close();
+
+        // # Step 3: Search the index
+        // Open a DirectoryReader to read the index
+        DirectoryReader reader = DirectoryReader.open(directory);
+
+        // Create an IndexSearcher to perform searches on the indexed data
+        IndexSearcher searcher = new IndexSearcher(reader);
+
+        // # Step 4: Build the CombinedFieldQuery
+        // Create a CombinedFieldQuery builder that allows us to query across multiple fields
+        CombinedFieldQuery.Builder queryBuilder = new CombinedFieldQuery.Builder();
+
+        // Add the fields we want to query: year, make, and model
+        queryBuilder.addField("year");
+        queryBuilder.addField("make");
+        queryBuilder.addField("model");
+
+        // Add terms for the specific query (e.g., "2011 Toyota Corolla")
+        queryBuilder.addTerm(new BytesRef("2011")); // Query for the year as a string
+        queryBuilder.addTerm(new BytesRef("Toyota")); // Query for the make
+        queryBuilder.addTerm(new BytesRef("Corolla")); // Query for the model
```

Comment:
If you check out the constructor for `CombinedFieldQuery` (which gets called from `CombinedFieldQuery.Builder#build()`), there's this part here:

```
    for (String field : fieldAndWeights.keySet()) {
      for (BytesRef term : terms) {
        fieldTerms[pos++] = new Term(field, term);
      }
    }
```

Essentially, it adds a `Term` for every field/term combination.

I think it would be good to run the following query (which would match the same set of docs) as well, and show how the scores are different:

```
        BooleanQuery.Builder boolQueryBuilder = new BooleanQuery.Builder();
        for (String field : List.of("year", "make", "model")) {
            for (String term : List.of("2011", "Toyota", "Corolla")) {
                boolQueryBuilder.add(new TermQuery(new Term(field, term)), BooleanClause.Occur.SHOULD);
            }
        }
        BooleanQuery boolQuery = boolQueryBuilder.build();
```

(A similar query is created in the `private rewriteToBoolean` method in `CombinedFieldQuery`.)

Reply:
Thanks @msfroh:

- Added better data with `author`, `title`, and `description`.
- Compared the scores with `boolQueryBuilder`,  `combinedFieldQueryBuilder` and `docCombinedFieldQuery` (the combined field).

Now I see that the `docCombinedFieldQuery` and` combinedFieldQueryBuilder` produce the same scores because `combinedFieldQueryBuilder` treats the fields as one virtual field, similar to how `combined_field` is used in `docCombinedFieldQuery`.